lib/deployment_runner.py

```diff
@@ -110,6 +123,186 @@ def rhv_storage(self, page):
             pass
         return page.click_next()
 
+    def osp_detect_undercloud(self, page):
+        '''OSP Detect Undercloud Page filler'''
+        page.set_undercloud_ip(self.osp.undercloud_address)
+        page.set_ssh_user(self.osp.undercloud_user)
+        page.set_ssh_password(self.osp.undercloud_pass)
+        page.click_detect_undercloud()
+        return page.click_next()
+
+    def osp_register_nodes(self, page):
+        '''
+        OSP Register Nodes Page Filler
+
+        This one is pretty complicated.   First we have to click register
+        nodes button.   This opens a modal, where one can fill in various ways
+        of selecting our nodes.   Presently, we only support doing autodetect.
+        Information to do autodetec is filled in and then Next is selected, but
+        this takes you to a new modal where you select which machines to
+        register.
+
+        TODO: The yaml and software supports registering nodes from multiple
+              Hosts, however this code only supports one Host presently.
+              At some point we should make it support multiple hosts.
+        '''
+
+        ##################################
+        # The Register Nodes Modal Frame #
+        ##################################
+        for node_registration_info in self.osp.overcloud_nodes:
+            page.click_register_nodes()
+            self._osp_determine_how_to_register_nodes(
+                page,
+                node_registration_info
+            )
+
+        # Wait for the spinners to be done:
+        # XXX: I thought sometimes the wait would kick off before
+        # the spinners started, so we have this arbitrary sleep.
+        # We might be able to remove this.
+        time.sleep(3)
+        page.wait_on_spinner(
+            timeout=300,
+            spin_class='spinner-xs'
+        )
+
+        return page.click_next()
+
+    def _osp_determine_how_to_register_nodes(self, page, node_registration_info):
```

Comment:
Instead of a monolithic method for registering nodes, would it be beneficial to have 3 separate methods for each type of registration (manual, autodetect, csv)?  It would allow future additions for autodetect & csv while minimizing the risk of introducing bugs in to existing functionality.